### PR TITLE
Additional failure tests

### PR DESCRIPTION
**Pull request by jmorel**
We were only testing the failure behavior for traintuples. This adds tests for aggregatuples and composite traintuples as well.

### CODE REVIEW

File: tests/test_execution.py [hunk 1]
```diff
@@ -149,6 +149,63 @@ def test_traintuple_execution_failure(global_execution_env):
...
+        composite_traintuples.append(session.add_composite_traintuple(spec))
+
+    spec = factory.create_aggregatetuple(
+        algo=aggregate_algo,
+        traintuples=composite_traintuples,
+        worker=session.node_id,
+    )
+    aggregatetuple = session.add_aggregatetuple(spec).future().wait(raises=False)
+    for composite_traintuple in composite_traintuples:
+        assert session.get_composite_traintuple(composite_traintuple.key).status == assets.Status.done
```
Comment (samlesu) [inline tests/test_execution.py]: Really minor: could be done in two lines.

Comment (jmorel) [reply tests/test_execution.py]: You mean no for loop and two assertions or just break the L204 into 2 lines?

Comment (samlesu) [reply tests/test_execution.py]: Sorry for the confusion,  just breaking the L204 into 2 line.
But that's so minor that I approved the PR, so up to you to do the change or not. :)

File: tests/test_execution.py [hunk 2]
```diff
@@ -149,6 +149,64 @@ def test_traintuple_execution_failure(global_execution_env):
...
+
+    spec = factory.create_aggregatetuple(
+        algo=aggregate_algo,
+        traintuples=composite_traintuples,
+        worker=session.node_id,
+    )
+    aggregatetuple = session.add_aggregatetuple(spec).future().wait(raises=False)
+    for composite_traintuple in composite_traintuples:
+        composite_traintuple = session.get_composite_traintuple(composite_traintuple.key)
+        assert composite_traintuple.status == assets.Status.done
```
Comment (AurelienGasser) [inline tests/test_execution.py]: Minor: considering that the test's purpose is to ensure the aggtuple fails, do we need this block that ensures composite traintuples succeeded?

Comment (jmorel) [reply tests/test_execution.py]: If one of the composite traintuple failed, the aggregatuple would also automatically fail. This makes sure the reason the aggregatetuple failed is its invalid code and not an invalid dependency.

**Pull request merged**